drivers/SmartThings/matter-sensor/src/test/test_matter_smoke_co_alarm.lua

```diff
@@ -73,18 +74,20 @@ local cluster_subscribe_list = {
 }
 
 local function test_init()
+  -- The startup messages are enabled, so this device will get an init,
+  -- and doConfigure (because provisioing_state is TYPED on the device).
+  test.mock_device.add_test_device(mock_device)
```

Comment:
We dont need to simulate the whole added/init/doConfigure flow since the driver isn't doing anything special (namely setting fields on the device object); the test framework will queue init and doConfigure automatically, which is why they aren't queued here, but we still are setting expectations on the subscribe and update to provisioning state.